### ISSUE TITLE
[BUG] Submit an idea is not working as expected.

### ISSUE THREAD

**Issue by vidisha-qa** [dev-launchers/dev-launchers-platform#973]
[https://github.com/dev-launchers/dev-launchers-platform/issues/787#issue-1512290131](url) 
**Summary**: Submit an idea is not working as expected.

**Scenario** 1: 
**Description**: Idea is not getting submited only after filing up mandatory fields. Wait button get displayed and no other actions happen even after waiting for long time.

**Steps**:
Pre-requisite : User should be logged-in into dev launchers.
step-1: Navigate to URL "https://staging.devlaunchers.org/ideaspace/submit".
step-2: Enter only all mandatory field.
step-3: Check the checkbox for 'Terms & Condition'.
step-4: Click on Submit button.

**Actual Output**: After filling up all mandatory fields of 'Submit an idea' form and clicking on Submit button user is not getting navigate to Idea workshop page and "Wait" button get replaced by "Submit" button. 

**Expected Output**: After filling up all mandatory fields of 'Submit an idea' form and clicking on Submit button then user should be redirected to Idea workshop page. 

**Scenario** 2: Error gets displayed on submitting Idea after filing up all the fields. 

**Steps**:
Pre-requisite : User should be logged-in into dev launchers.
step-1: Navigate to URL "https://staging.devlaunchers.org/ideaspace/submit".
step-2: Enter all the fields.
step-3: Check the checkbox for 'Terms & Condition'.
step-4: Click on Submit button.

**Actual Output**: After filling up all fields of 'Submit an idea' form and clicking on Submit button user is navigated to Idea workshop page with no proper content and Application error displayed. (idea-card API status: 500)

**Expected Output**: After filling up all mandatory fields of 'Submit an idea' form and clicking on Submit button then user should be redirected to Idea workshop page. 

https://user-images.githubusercontent.com/79625447/228284502-fe9116e2-c218-41b7-a0d4-ebe0ead1ecb7.mp4

https://user-images.githubusercontent.com/79625447/228284652-e31f0f9e-f654-484e-baf6-b48dcec5aceb.mp4

**Comment by alicefate** [comment]
@dbradham  How can I log in staging website? The login button in navbar and page body all redirect to production.

**Comment by dbradham** [comment]
@alicefate you have to delete your cookies or use a different browser to log in to the staging. (and then you'll have to do the same thing to get back into production)

**Comment by dbradham** [comment]
@alicefate You can reproduce this at the following link: https://staging.devlaunchers.org/ideaspace/workshop/230

The problem is that the username is not being passed in correctly, and the workshopping page is not handling null inputs in a graceful way so the whole page breaks. We should review how the username is being set in the input form for this ticket. 

![image.png](https://images.zenhubusercontent.com/639ba80c96c787af67163973/001477a8-3769-4295-abdb-751a96185677)

**Comment by alicefate** [comment]
@dbradham username is from author, and author has been added from backend instead of frontend. Since the new IdeaCard Schema has been approved. We also need issue 292 to be implemented as soon as possible.

**Comment by JulieMass** [comment]
Hey all @vidisha-qa, @dbradham and @alicefate, 

I've tried to reproduce this bug and the form in staging seems to be working now. But I'm confused with the comments here.  Is there still work to be done for this ticket or is it fixed? 

https://images.zenhubusercontent.com/225525029/b487386f-47cf-406f-a91a-696525d87619/_bug__submit_an_idea_is_not_working_as_expected___13_april_2023.mp4

**Comment by alicefate** [comment]
@JulieMass In video, it's production area not stagging. The quik way to figure out is button has no color in production.

**Comment by JulieMass** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/IdeaSpace-Team-6350185d9843157427cc900b/issues/dev-launchers/dev-launchers-platform/973?planning-poker) with Zenhub @alicefate @dbradham @jordanmcousins @JulieMass @katehirschman @Nouri-Anouar @rashikashaw @RETprojects

**Comment by JulieMass** [comment]
Hey @dbradham, I'm a bit confused about what happened with this ticket. It was assigned to you, but when I checked the history, it seems that Iris was working on it. Also, my only record of the assignment is in our sprint 7, but since then, it seems like we forgot about it 😅.

Anyway, **could you please help me update the ticket's final estimation and status?** We need to make sure this is fixed because Eliana reported this week that she wasn't able to submit an idea in production. 

Thanks a lot!

**Comment by dbradham** [comment]
I confirmed we still have an issue here linking ideas to users @JulieMass. This is completely different than Eliana's reported issue. (not sure what is causing that right now but I know it will be resolved with our newest version's deployment)

I'm working on this now as well as fixing the "updated at" field on the idea cards on the browse page.

**Comment by dbradham** [comment]
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZ0h6IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--d7a883aacb2ae209b272e232e64f5d580a0ffc16/image.png)

The issue Eliana reported could be related to the backend validations. These are not presented to the user. We may want a new ticket to present these validation failures to a user. Apparently they cannot use a name for their idea that has already been used.

**Comment by dbradham** [comment]
Thread in discord which describes backend issues I am having here.

There was a problem here, we were passing in the entire object for the user (JSON from the userDataContext) instead of the user id. But even after changing to use the user id, I am not seeing any difference.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZzN6IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ae17dc7ec1aafe0ea619e16a6aa86e1940e510c2/image.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZzd6IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--42136ca3809ce3f96d5d1c11caa3a7ec60825667/image.png)
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZy96IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--91502eaa5ef4c7287c1035ee84de2ecc2fde51f7/image.png)

**Comment by JulieMass** [comment]
Hey @dbradham I tried submitting an idea through staging and I wasn't able so we can't consider it closed. Any idea why? is this ticket related to the author issue that Remi has been working on lately?

**Comment by dbradham** [comment]
@JulieMass can you post what you are seeing in staging? I am not able to load any pages in staging-<img width="1346" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN3M5QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bfaa6ae001a23171b3298d23ac98698d871e9cd4/image.png" alt="image.png" />

**Comment by dbradham** [comment]
@JulieMass I also confirmed the expected output is the behavior in production at devlaunchers.org

**Comment by JulieMass** [comment]
> @JulieMass I also confirmed the expected output is the behavior in production at devlaunchers.org

Hey @dbradham I checked the production and staging environments, and I'm having trouble submitting an idea in both places. I've included some screenshots below to show you what's going on.

Production:
![image](https://github.com/dev-launchers/dev-launchers-platform/assets/104226366/6a2de3e7-acc3-433c-86a3-4c8df1bc4d29)

Staqing:
![image](https://github.com/dev-launchers/dev-launchers-platform/assets/104226366/344920be-f4f7-4783-8c1e-3900690e15f4)

**Comment by JulieMass** [comment]
Closing this ticket given that we can now submit ideas in global staging. 

Thank you @dbradham again!

(CC. @chadcrotchett)
